src/uu/touch/src/touch.rs

```diff
@@ -126,7 +126,20 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
             }
 
             if let Err(e) = File::create(path) {
-                show!(e.map_err_context(|| format!("cannot touch {}", path.quote())));
+                let is_directory = if let Some(last_char) = path.to_string_lossy().chars().last() {
```

Comment:
path.is_dir() doesn't work here ?

Reply:
I understand the idea, but no. Because `path.is_dir()` call fs::metadata under the hood (and check if it exits)

https://doc.rust-lang.org/src/std/path.rs.html#2949-2951

Reply:
please add a comment then :)

Reply:
yes! added!